radosgw_agent/worker.py

```diff
@@ -292,7 +344,11 @@ def get_bucket_instance_entries(self, marker, instance):
 
     def inc_sync_bucket_instance(self, instance, marker, timestamp, retries):
         max_marker, entries = self.get_bucket_instance_entries(marker, instance)
-        objects = set([entry.object for entry in entries])
+
+        # regardless if entries are versioned, make sure we filter them
+        entries = [i for i in ifilter(filter_versioned_objects, entries)]
+
+        objects = set([entry for entry in entries])
```

Comment:
we are no longer using `entry.object` because we now need the extra information about the object itself if the object is versioned. That is why `list_objects_in_bucket` no longer yields the key but the actual object.

In `sync_object_intra_region` you can see that we use those extra attributes so that the logic for versioned objects happens only there.

What do you see as a potential issue here of using the whole object vs `entry.object` ?